### PR TITLE
feat(paymaster-proxy): create db schemas

### PR DESCRIPTION
**Pull request by jakim929**
adds a `sponsorshipPolicies` table.

The purpose of this table is to store a mapping from **api-key** (issued by api-key-service) to a **provider-specific policy** (ie. on Alchemy, Pimlico, Stackup)

- still have not added runtime migrations

### CODE REVIEW

**Comment by jakim929** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/278?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#278** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/278?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#268** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#260** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#256** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#255** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/255?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#230** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/230?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/paymaster-proxy/src/models/sponsorshipPolicies.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      .defaultNow()
+      .notNull(),
+    updatedAt: timestamp('updated_at', { withTimezone: true })
+      .defaultNow()
+      .notNull(),
+  },
+  (table) => {
+    return {
+      apiKeyIdIdx: uniqueIndex().on(table.apiKeyId), // one policy per api key
+      chainIdIdx: index().on(table.chainId),
```
Comment (jakim929) [inline apps/paymaster-proxy/src/models/sponsorshipPolicies.ts:45]: useful for admin functionality (checking all policies for a given chainId)

File: apps/paymaster-proxy/src/models/sponsorshipPolicies.ts
Comment (tremarkley) [inline apps/paymaster-proxy/src/models/sponsorshipPolicies.ts:1]: do you need to run the migrations for this? I dont see any .sql files for creating this table

**Review by tremarkley — APPROVED** [review]

File: apps/paymaster-proxy/src/db/Database.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+export function connectToDatabase(options: DatabaseOptions): Database {
+  const pool = new Pool(options)
+  return drizzle(pool)
+}
+
+/**
+ * This is a temporary function until we get our utility pod setup to run migrations
+ */
+export async function runMigrations(
```
Comment (tremarkley) [inline apps/paymaster-proxy/src/db/Database.ts:31]: can be in a separate pr but dont forget to add a call to this in the service initialization! you will also want to wrap it in a retry + backoff since there is a race condition between the service starting up and the sql server being available to connect to

**Review by tremarkley — APPROVED** [review]

File: apps/paymaster-proxy/src/models/sponsorshipPolicies.ts
Comment (jakim929) [reply apps/paymaster-proxy/src/models/sponsorshipPolicies.ts:1]: yeah not yet - will add another PR for doing runtime migrations. 

There's still a lot of work to be done to stand up the postgres instances so holding off on doing migrations until that's ready
